### ISSUE TITLE
Explain Backpropagation

### ISSUE THREAD

**Issue by vuhung16au** [vuhung16au/nlp-learning-journey#110]
Explain Backpropagation (as in NLP)
and save your response to `docs/backpropagation.md`

Highlight the points
- Why it is important to understand NLP
- What machine learning models (such as `Gradient Descent`) and NN models use Backpropagation
